### PR TITLE
WIP save mail message after failed scenario

### PR DESCRIPTION
**Pull request by fureev**
And revert milliseconds in config

### CODE REVIEW

File: src/Staffim/Behat/MailExtension/Context/MailContext.php [hunk 1]
```diff
@@ -26,11 +26,11 @@ public function iShouldSeeNewMailMessagesAfterTime($count, $time)
      */
     public function iShouldSeeNewMailMessagesAfterWaiting($count)
     {
-        $sleepTime = $this->getMailAgentParameters()['maxSleepTime'];
-        if (!$this->getMailAgent()->wait($sleepTime  * 1000, $count)) {
+        $sleepTime = $this->getMailAgentParameters()['maxDuration'];
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/MailContext.php]: What about PHP 5.3?

**Comment by alexeyshockov** [comment]
How expecting a messages are corresponding to current active (viewed) mail? This isn't clear for me.

**Comment by alexeyshockov** [comment]
Give me use case.

**Comment by fureev** [comment]
Use case:

> И открываю письмо "subj"
> И должен видеть "text" в письме

On fail save mail message with name: scenario:line<mailTo>:mailSubj 
in config path dir if it isn't null.

**Comment by alexeyshockov** [comment]
OK, this usage is clear for me. Useful. Thanks!

File: src/Staffim/Behat/MailExtension/Context/RawMailContext.php [hunk 2]
```diff
@@ -70,11 +73,34 @@ public function getMail()
     /**
      * Return parameters provided for MailAgent.
      *
-     * @return array $parameters
+     * @param null $key
+     * @return mixed $parameters
      */
-    public function getMailAgentParameters()
+    public function getMailAgentParameters($key = null)
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/RawMailContext.php]: Split this method to separated getMailAgentParameters() and getMailAgentParameter().

P.S. getMailAgentParameter() can return Option ;)

File: src/Staffim/Behat/MailExtension/Context/MailContext.php [hunk 3]
```diff
@@ -26,11 +26,11 @@ public function iShouldSeeNewMailMessagesAfterTime($count, $time)
      */
     public function iShouldSeeNewMailMessagesAfterWaiting($count)
     {
-        $sleepTime = $this->getMailAgentParameters()['maxSleepTime'];
-        if (!$this->getMailAgent()->wait($sleepTime  * 1000, $count)) {
+        $sleepTime = $this->getMailAgentParameters('maxDuration');
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/MailContext.php]: `getMailAgentParameter()`?

File: src/Staffim/Behat/MailExtension/Context/MailContext.php [hunk 4]
```diff
@@ -131,7 +131,7 @@ public function iFillInFromMailValue($field, $pattern)
     public function iShouldSeeServerAddressInFrom($arg1)
     {
         return array(
-            new Step\When(sprintf('should see "%s" as reply address in mail message', $this->getMailAgentParameters()["baseAddress"])),
+            new Step\When(sprintf('should see "%s" as reply address in mail message', $this->getMailAgentParameters("baseAddress"))),
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/MailContext.php]: `getMailAgentParameter()`?

File: src/Staffim/Behat/MailExtension/Context/RawMailContext.php [hunk 5]
```diff
@@ -70,7 +74,7 @@ public function getMail()
     /**
      * Return parameters provided for MailAgent.
      *
-     * @return array $parameters
+     * @return mixed $parameters
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/RawMailContext.php]: Why not array?

File: src/Staffim/Behat/MailExtension/Context/RawMailContext.php [hunk 6]
```diff
@@ -78,6 +82,41 @@ public function getMailAgentParameters()
...
+     * Return parameters provided for MailAgent.
+     *
+     * @param string $key
+     * @throws \Symfony\Component\Config\Definition\Exception\Exception
+     * @return mixed $parameter
+     */
+    public function getMailAgentParameter($key)
+    {
+        if (!isset($this->mailAgentParameters[$key])) {
+
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/RawMailContext.php]: Unnecessary blank line.

File: src/Staffim/Behat/MailExtension/Context/RawMailContext.php [hunk 7]
```diff
@@ -78,6 +82,41 @@ public function getMailAgentParameters()
...
+            throw new Exception("Parameter doesn't exist");
+        }
+        $parameter = $this->mailAgentParameters[$key];
+
+        return $parameter;
```
Comment (alexeyshockov) [inline src/Staffim/Behat/MailExtension/Context/RawMailContext.php]: Unnecessary temporary variable.

**Pull request merged**